src/main.lua

```diff
@@ -420,6 +420,16 @@ end
 
 -- Sanitize inputs before every interaction
 local function assertAndSanitizeInputs(msg)
+	if msg.Tags.Action ~= "Eval" and msg.Data then
+		-- assert data size is less than or equal to 100 bytes, we dont use msg.Data
+
+		if type(msg.Data) == "table" then
+			msg.Data = json.encode(msg.Data)
```

Comment:
the json.encode here with a large data size will kill us - we want to essentially avoid doing this entirely.

Reply:
42b1432

Reply:
> the json.encode here with a large data size will kill us - we want to essentially avoid doing this entirely.

nit: If it is a large data size and we throw then it shouldn't matter, no? JSON encoding sub 100 bytes is trivial, might be worth it for simplicity?

Reply:
fwiw I believe json.encode happens within the AO code -- but our size checking function works here fine